haystack/core/serialization.py

```diff
@@ -11,6 +11,14 @@ def component_to_dict(obj: Any) -> Dict[str, Any]:
     """
     The marshaller used by the Pipeline. If a `to_dict` method is present in the
```

Comment:
I see now how this could be confusing because recently we introduced the `Marshaller` concept. I would rephrase these two docstrings with something along the lines of "Converts a component instance into a dictionary.."